sdk/core/azure-core/src/main/java/com/azure/core/implementation/http/rest/RestProxyUtils.java

```diff
@@ -95,4 +113,115 @@ private static Flux<ByteBuffer> validateFluxLength(Flux<ByteBuffer> bbFlux, long
             });
         });
     }
+
+    /**
+     * Validates the Length of the input request matches its configured Content Length.
+     * @param request the input request to validate.
+     * @return the requests body as BinaryData on successful validation.
+     */
+    public static BinaryData validateLengthSync(final HttpRequest request) {
+        final BinaryData binaryData = request.getBodyAsBinaryData();
+        if (binaryData == null) {
+            return binaryData;
+        }
+
+        final long expectedLength = Long.parseLong(request.getHeaders().getValue("Content-Length"));
+        Long length = binaryData.getLength();
+        BinaryDataContent bdc = BinaryDataHelper.getContent(binaryData);
+        if (length == null) {
+            if (bdc instanceof FluxByteBufferContent) {
+                throw new IllegalStateException("Flux Byte Buffer is not supported in Synchronous Rest Proxy.");
+            } else if (bdc instanceof InputStreamContent) {
+                InputStreamContent inputStreamContent = ((InputStreamContent) bdc);
+                InputStream inputStream = inputStreamContent.toStream();
+                LengthValidatingInputStream lengthValidatingInputStream =
+                    new LengthValidatingInputStream(inputStream, expectedLength);
+                return BinaryData.fromStream(lengthValidatingInputStream);
```

Comment:
we should get the bug fix made for async version here as well.
I.e. we should attach length validating stream to InputStreamContent regardless of length being null or not.